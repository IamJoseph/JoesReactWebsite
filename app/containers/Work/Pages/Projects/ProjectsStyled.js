import styled from 'styled-components';
import nasa from '../../../../assets/nasa.jpg';

export const H2 = styled.h2`
  background: -webkit-linear-gradient(transparent, transparent), url(${nasa});
  background-size: contain;
  color: #686664;
  font-family: sans-serif;
  font-size: 1.9em;
  letter-spacing: 1px;
  line-height: normal;
  margin-bottom: 0;
  text-align: center;
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  -webkit-text-stroke: 1px;
  word-break: break-word;
  @media(min-width: 768px){
    font-size: 2.5em;
    flex: 1 100%;
  }
  span{
    -webkit-text-fill-color: currentColor;
    -webkit-text-stroke-width: 0;
  }
`;

export const UL = styled.ul`
  align-items: center;
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  line-height: 2.5em;
  list-style: none;
  max-width: 100%;
  padding: 0;
  @media(min-width: 768px){
    flex-direction: row;
    justify-content: center;
  }
`;

export const Div = styled.div`
  padding: 2em 1em;
  @media(min-width: 768px){
    flex: 1 50%;
    order: 3;
  }
`;

export const I = styled.i`
  line-height: normal;
  margin-bottom: 3em;
  text-align: center;
  @media(min-width: 768px){
    flex: 1 100%;
    order: 2;
  }
`;

export const ImgContainer = styled.div`
  max-width: 100%;
  @media(min-width: 768px){
    flex: 1 50%;
    margin-top: 3em;
    max-width: 50%;
  }
  &.clicked{
    background-color: #fff;
    bottom: 0;
    height: 100%;
    left: 0;
    margin: 0;
    max-width: 100%;
    overflow-y: auto;
    padding: 0 25px;
    position: fixed;
    right: 0;
    text-align: center;
    top: 0;
    width: 100%;
    img{
      margin-top: 80px;
      max-width: 1460px;
      width: 100%;
    }
  }
`;

export const IMG = styled.img`
  box-shadow: 0px 12px 12px -10px #3a3a3a;
  cursor: pointer;
  max-width: 100%;
  @media(min-width: 768px){
    max-width: 95%;
  }
`;

export const VideoContainer = styled.div`
  max-height: 550px;
  max-width: 80%;
  text-align: center;
  @media(min-width: 768px){
    flex: 1 50%;
    height: 380px;
    margin-top: 3em;
    max-width: 50%;
    order: 2;
  }
  @media(min-width: 1000px){
    height: 550px;
  }
  &.clicked{
    background-color: #fff;
    bottom: 0;
    height: 100%;
    left: 0;
    margin: 0;
    max-height: 100%;
    max-width: 100%;
    position: fixed;
    right: 0;
    text-align: center;
    top: 0;
    width: 100%;
    video{
      margin-top: 80px;
      max-height: calc(100% - 80px);
      max-width: 1460px;
      padding: 0 25px;
      width: 100%;
    }
  }
`;

export const Video = styled.video`
  border-radius: 3px;
  box-shadow: 0px 12px 12px -10px #3a3a3a;
  cursor: pointer;
  max-height: 550px;
  max-width: 100%;
`;
