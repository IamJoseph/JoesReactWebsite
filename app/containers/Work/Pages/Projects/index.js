import React, { PureComponent } from 'react';
import { Container, AWrapper, LI } from '../PagesStyled';
import { UL, H2, I, Div, Video, VideoContainer, IMG, ImgContainer } from './ProjectsStyled';
import farmboyInv from '../../../../assets/farmboy_inventory.gif';
import borkVid from '../../../../assets/Bork_720.mp4';
import toggle from '../toggleFullScreen';

export default class Projects extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      bork: false,
      inventory: false,
    };
  }

  toggleFullScreen = toggle;

  render() {
    const { bork, inventory } = this.state;
    return (
      <Container>
        <UL>
          <H2 style={{ fontSize: '3em' }}>
            Bork <span role="img" aria-label="dog">&#128054;</span>
          </H2>
          <I>for iOS and Android</I>
          <Div>
            <AWrapper>
              <a rel="dns-prefetch" href="https://github.com/IamJoseph/Bork" target="_blank">
              Bork @ GitHub
            </a>
            </AWrapper>
            <LI>
              react-native / redux / redux-thunk / react-router-native-flux / babel /
              webpack / react-native-maps / react-native-camera / firebase
            </LI>
            <LI>
              Pure es6/esNext implementation
            </LI>
            <em>
              Click the GitHub link for more details!
            </em>
          </Div>
          <VideoContainer onClick={() => this.toggleFullScreen('bork')} className={bork && 'clicked'}>
            <Video autoPlay muted playsInline loop src={borkVid} />
          </VideoContainer>
        </UL>
        <UL>
          <H2>
            Farmboy Inventory
          </H2>
          <Div>
            <AWrapper>
              <a rel="dns-prefetch" href="https://github.com/IamJoseph/InventoryGUI" target="_blank">
              InventoryGUI @ GitHub
            </a>
            </AWrapper>
            <LI>
              MySQLi / php / jquery / jquery-ui
            </LI>
          </Div>
          <ImgContainer onClick={() => this.toggleFullScreen('inventory')} className={inventory && 'clicked'}>
            <IMG src={farmboyInv} alt="farmboy inventory application" />
          </ImgContainer>
        </UL>
      </Container>
    );
  }
}
