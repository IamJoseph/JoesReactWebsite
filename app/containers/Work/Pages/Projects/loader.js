/**
 * Asynchronously loads the components for WorkPage
 */

import { errorLoading } from 'utils/asyncInjectors';

export default () => (cb) => {
  import('containers/Work/Pages/Projects')
    .then(cb)
    .catch(errorLoading);
};
