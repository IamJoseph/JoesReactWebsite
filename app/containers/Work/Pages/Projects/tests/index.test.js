import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Projects from '../index';
import { VideoContainer, ImgContainer } from '../ProjectsStyled';

describe('<Projects />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <Projects />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should toggle className clicked on video container when clicked', () => {
    const wrapper = shallow(<Projects />);
    wrapper.find(VideoContainer).simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should toggle className clicked on image container when clicked', () => {
    const wrapper = shallow(<Projects />);
    wrapper.find(ImgContainer).simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
