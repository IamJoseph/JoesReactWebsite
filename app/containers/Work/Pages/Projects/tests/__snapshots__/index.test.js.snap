// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Projects /> should render correctly 1`] = `
<styled.section>
  <styled.ul>
    <styled.h2
      style={
        Object {
          "fontSize": "3em",
        }
      }
    >
      Bork 
      <span
        aria-label="dog"
        role="img"
      >
        üê∂
      </span>
    </styled.h2>
    <styled.i>
      for iOS and Android
    </styled.i>
    <styled.div>
      <styled.div>
        <a
          href="https://github.com/IamJoseph/Bork"
          rel="dns-prefetch"
          target="_blank"
        >
          Bork @ Github
        </a>
      </styled.div>
      <styled.li>
        react-native / redux / redux-thunk / react-router-native-flux / babel /
        <br />
        webpack / react-native-maps / react-native-camera / firebase
      </styled.li>
      <styled.li>
        Pure es6/esNext implementation
      </styled.li>
    </styled.div>
    <styled.div
      className={false}
      onClick={[Function]}
    >
      <styled.video
        autoPlay={true}
        loop={true}
        muted={true}
        playsInline={true}
        src="IMAGE_MOCK"
      />
    </styled.div>
  </styled.ul>
  <styled.ul>
    <styled.h2>
      Farmboy Inventory
    </styled.h2>
    <styled.div>
      <styled.div>
        <a
          href="https://github.com/IamJoseph/InventoryGUI"
          rel="dns-prefetch"
          target="_blank"
        >
          InventoryGUI @ Github
        </a>
      </styled.div>
      <styled.li>
        MySQLi / php / jquery / jquery-ui
      </styled.li>
    </styled.div>
    <styled.div
      className={false}
      onClick={[Function]}
    >
      <styled.img
        alt="farmboy inventory application"
        src="IMAGE_MOCK"
      />
    </styled.div>
  </styled.ul>
</styled.section>
`;

exports[`<Projects /> should toggle className clicked on image container when clicked 1`] = `
<styled.section>
  <styled.ul>
    <styled.h2
      style={
        Object {
          "fontSize": "3em",
        }
      }
    >
      Bork 
      <span
        aria-label="dog"
        role="img"
      >
        üê∂
      </span>
    </styled.h2>
    <styled.i>
      for iOS and Android
    </styled.i>
    <styled.div>
      <styled.div>
        <a
          href="https://github.com/IamJoseph/Bork"
          rel="dns-prefetch"
          target="_blank"
        >
          Bork @ Github
        </a>
      </styled.div>
      <styled.li>
        react-native / redux / redux-thunk / react-router-native-flux / babel /
        <br />
        webpack / react-native-maps / react-native-camera / firebase
      </styled.li>
      <styled.li>
        Pure es6/esNext implementation
      </styled.li>
    </styled.div>
    <styled.div
      className={false}
      onClick={[Function]}
    >
      <styled.video
        autoPlay={true}
        loop={true}
        muted={true}
        playsInline={true}
        src="IMAGE_MOCK"
      />
    </styled.div>
  </styled.ul>
  <styled.ul>
    <styled.h2>
      Farmboy Inventory
    </styled.h2>
    <styled.div>
      <styled.div>
        <a
          href="https://github.com/IamJoseph/InventoryGUI"
          rel="dns-prefetch"
          target="_blank"
        >
          InventoryGUI @ Github
        </a>
      </styled.div>
      <styled.li>
        MySQLi / php / jquery / jquery-ui
      </styled.li>
    </styled.div>
    <styled.div
      className="clicked"
      onClick={[Function]}
    >
      <styled.img
        alt="farmboy inventory application"
        src="IMAGE_MOCK"
      />
    </styled.div>
  </styled.ul>
</styled.section>
`;

exports[`<Projects /> should toggle className clicked on video container when clicked 1`] = `
<styled.section>
  <styled.ul>
    <styled.h2
      style={
        Object {
          "fontSize": "3em",
        }
      }
    >
      Bork 
      <span
        aria-label="dog"
        role="img"
      >
        üê∂
      </span>
    </styled.h2>
    <styled.i>
      for iOS and Android
    </styled.i>
    <styled.div>
      <styled.div>
        <a
          href="https://github.com/IamJoseph/Bork"
          rel="dns-prefetch"
          target="_blank"
        >
          Bork @ Github
        </a>
      </styled.div>
      <styled.li>
        react-native / redux / redux-thunk / react-router-native-flux / babel /
        <br />
        webpack / react-native-maps / react-native-camera / firebase
      </styled.li>
      <styled.li>
        Pure es6/esNext implementation
      </styled.li>
    </styled.div>
    <styled.div
      className="clicked"
      onClick={[Function]}
    >
      <styled.video
        autoPlay={true}
        loop={true}
        muted={true}
        playsInline={true}
        src="IMAGE_MOCK"
      />
    </styled.div>
  </styled.ul>
  <styled.ul>
    <styled.h2>
      Farmboy Inventory
    </styled.h2>
    <styled.div>
      <styled.div>
        <a
          href="https://github.com/IamJoseph/InventoryGUI"
          rel="dns-prefetch"
          target="_blank"
        >
          InventoryGUI @ Github
        </a>
      </styled.div>
      <styled.li>
        MySQLi / php / jquery / jquery-ui
      </styled.li>
    </styled.div>
    <styled.div
      className={false}
      onClick={[Function]}
    >
      <styled.img
        alt="farmboy inventory application"
        src="IMAGE_MOCK"
      />
    </styled.div>
  </styled.ul>
</styled.section>
`;
